draft-yasskin-http-origin-signed-responses.md

```diff
@@ -941,35 +956,37 @@ Because different clients send different request header fields, and intermediate
 servers add response header fields, it can be impossible to have a signature for
 the exact request and response that the client sees. Therefore, when a client
 validates the `Signature` header field for an exchange represented as a normal
-HTTP request/response pair, it MUST pass only the subset of header fields
-defined by {{significant-headers}} to the validation procedure
-({{signature-validity}}).
+HTTP request/response pair, it MUST pass the serialized headers defined by
+{{serialized-headers}} to the validation procedure ({{signature-validity}})
+along with the `Signature` header field and the response's payload.
 
 If the client relies on signature validity for any aspect of its behavior, it
 MUST ignore any header fields that it didn't pass to the validation procedure.
 
-### Significant headers for a same-origin response {#significant-headers}
+### Serialized headers for a same-origin response {#serialized-headers}
 
-The significant headers of an exchange represented as a normal HTTP
+The serialized headers of an exchange represented as a normal HTTP
 request/response pair (Section 2.1 of {{?RFC7230}} or Section 8.1 of
-{{?RFC7540}}) are:
+{{?RFC7540}}) are the canonical serialization ({{canonical-cbor}}) of the CBOR
+representation ({{cbor-representation}}) of the following request and response
+metadata and headers:
 
 * The method (Section 4 of {{!RFC7231}}) and effective request URI (Section 5.5
   of {{!RFC7230}}) of the request.
 * The response status code (Section 6 of {{!RFC7231}}) and the response header
   fields whose names are listed in that exchange's `Signed-Headers` header field
-  ({{signed-headers}}), in the order they appear in that header field. If a
-  response header field name from `Signed-Headers` does not appear in the
-  exchange's response header fields, the exchange has no significant headers.
+  ({{signed-headers}}). If a response header field name from `Signed-Headers`
+  does not appear in the exchange's response header fields, the exchange has no
+  serialized headers.
```

Comment:
I get a bit confused by the language "no serialized headers". It sounds like this means :method, :url, and :status would be omitted from {{cbor-representation}}, but I don't think that's what's intended.

Reply:
I'll do a separate PR for this.